examples/21_create-pages/src/components/RoutingHandler.tsx

```diff
@@ -1,18 +1,18 @@
 'use client';
 
-import { useCallback } from 'react';
+import { useEffect } from 'react';
 import { useRouter } from 'waku/router/client';
 
 export const RoutingHandler = () => {
   const { unstable_onRouteChangeComplete, unstable_onRouteChangeStart } =
     useRouter();
-  unstable_onRouteChangeStart(
-    useCallback(() => {
-      console.log('onRouteChangeStart');
-    }, []),
-  );
-  unstable_onRouteChangeComplete(() => {
-    console.log('onRouteChangeComplete');
-  });
+  useEffect(() => {
+    unstable_onRouteChangeStart((...args) => {
+      console.log('onRouteChangeStart', args);
+    });
+    unstable_onRouteChangeComplete((...args) => {
+      console.log('onRouteChangeComplete', args);
+    });
+  }, [unstable_onRouteChangeComplete, unstable_onRouteChangeStart]);
```

Comment:
hm, if it looks like this, the better api would probably just be to give `useRouter().isPending` 

I personally like the `useRouter().events.on('routechangestart', someCallback)` api best from what I've seen and I think that would need to use emitted browser events to work?

Reply:
Actually, since I changed it to use a RefObject, it probably doesn't have to still go in a useEffect to avoid over-triggering.

Reply:
Yup. I was able to simplify that to remove `useEffect`.

Re: `useRouter().isPending` -- I think we don't want to add startTransition or state updates though. Not sure. The event callback approach won't add anything to React unless the developer decides to - which is probably nice for this use case.